source/ConditionsStore.cpp

```diff
@@ -459,3 +465,60 @@ bool ConditionsStore::VerifyProviderLocation(const string &name, DerivedProvider
 				", because it is within range of prefixed derived provider \"" + ce.provider->name + "\".");
 	return true;
 }
+
+
+
+// Expands the name of a condition to the full name based on the $[] blocks it contains.
+string ConditionsStore::Expand(const string &name) const
+{
+	using size_t = string::size_type;
```

Comment:
Isn't size_t already defined?
I don't understand why you do this.

Reply:
string::size_type is the type that strings are indexed by, which might or might not be the same as the global size_t. (Tbh I'm not sure if it can actually be different. I think it depends on the allocator?)

Reply:
From cppreference:

> std::size_t can store the maximum size of a theoretically possible object of any type (including array). A type whose size cannot be represented by std::size_t is ill-formed. On many platforms (an exception is systems with segmented addressing) std::size_t can safely store the value of any non-member pointer, in which case it is synonymous with std::uintptr_t.

> std::size_t is commonly used for array indexing and loop counting. Programs that use other types, such as unsigned int, for array indexing may fail on, e.g. 64-bit systems when the index exceeds UINT_MAX or if it relies on 32-bit modular arithmetic.

> When indexing C++ containers, such as std::string, std::vector, etc, the appropriate type is the member typedef size_type provided by such containers. It is usually defined as a synonym for std::size_t.

Reply:
In theory it can be different but in practice it isn't. Just use the global `size_t` :smile: